### DISCUSSION TITLE
Session expiration within MQTT 5.0

### DISCUSSION THREAD

**Discussion by fbrouille** [emqx/emqx#11566] [Q&A]
Hello, i found in documentation that EMQX provides the `mqtt.session_expiry_interval` configuration item to limit session duration within non-MQTT 5.0 connections.

I would like to know if `mqtt.session_expiry_interval` also applies to MQTT 5.0 connections (despite what the documentation says).
If not, how can i configure the broker to limit session duration for MQTT 5.0 connections in case a client specifies a session expiry interval very large or infinite (0xFFFFFFFF) ?
https://www.emqx.io/docs/en/v5.0/mqtt/mqtt-session-and-message-expiry.html#mqtt-persistent-sessions
https://www.emqx.io/docs/en/v5.0/configuration/configuration-manual.html#mqtt-basic-parameters

**Comment by savonarola** [comment]
Hello!
`session_expiry_interval` setting is used as the lifetime default for non-MQTTv5.0 connections.
Currently, there is no setting to _limit_ the lifetime specified by an MQTTv5.0 client.

**Reply by fbrouille** [reply]
Hello, thank you for your answer.
So if client is disconnected and can not connect again (because of network outages), does it mean lack of resources associated to the session on the broker side ?

**Reply by id** [reply]
@fbrouille depends. What error does client receive from the broker?
